js/ssai_pulse.js

```diff
@@ -237,19 +273,17 @@ OO.Ads.manager(function(_, $)
       {
         adMode = true;
         this.currentAd = ad;
-        if (this.currentAd.ad)
-        {
-          this.currentAd.ad.id3AdId = this.currentId3Object.adId;
+        if (ad.ad && ad.ad.data && (ad.duration && typeof ad.duration === 'number')) {
```

Comment:
@cacel I did some research on where this event goes, and I don't think we need to fail if ad.duration is not a number. The only place I found it used was in analytics as part of a metadata object. Instead of placing this validation check in the if statement (also use `_.isNumber` here for this), you should put it inside the if and assign some dummy value like 0 or -1 if the check fails